plat/arm/board/fvp/fvp_topology.c

```diff
@@ -16,15 +16,16 @@
 static unsigned char fvp_power_domain_tree_desc[FVP_CLUSTER_COUNT + 2];
 
 
-CASSERT(FVP_CLUSTER_COUNT && FVP_CLUSTER_COUNT <= 256, assert_invalid_fvp_cluster_count);
+CASSERT(((FVP_CLUSTER_COUNT > 0) && (FVP_CLUSTER_COUNT <= 256)),
+			assert_invalid_fvp_cluster_count);
 
 /*******************************************************************************
  * This function dynamically constructs the topology according to
  * FVP_CLUSTER_COUNT and returns it.
  ******************************************************************************/
 const unsigned char *plat_get_power_domain_tree_desc(void)
 {
-	unsigned int i;
+	int i;
```

Comment:
why is the type changed here ?

Reply:
variable "i" is used in the for loop and compared with FVP_CLUSTER_COUNT 2 (signed). This gets rid of a MISRA warning.

plat/arm/board/fvp/fvp_topology.c
36 MISRA C-2012 Rule 10.4 (required) 251345
Essential type of the left hand operand "i" (unsigned) is not the same as that of the right operand "2U"(signed).

plat/arm/board/fvp/fvp_topology.c
37 MISRA C-2012 Rule 10.4 (required) 251345
Essential type of the left hand operand "i" (unsigned) is not the same as that of the right operand "2U"(signed).